crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -340,6 +340,37 @@ impl<'w> EntityMut<'w> {
             })
         };
 
+        unsafe {
+            Self::move_entity_from_remove::<false>(
+                entity,
+                &mut self.location,
+                old_location.archetype_id,
+                old_location,
+                entities,
+                archetypes,
+                storages,
+                new_archetype_id,
+            );
+        }
+
+        Some(result)
+    }
+
+    /// when DROP is true removed components will be dropped otherwise they will be forgotten
+    #[allow(clippy::too_many_arguments)]
+    #[allow(clippy::match_bool)]
+    unsafe fn move_entity_from_remove<const DROP: bool>(
```

Comment:
The const generic bool is sweet. You should have a comment explaining how it's faster than just using a parameter though, otherwise you'll confuse devs seeing this for the first time.